crates/wasmtime/src/trap.rs

```diff
@@ -158,21 +152,22 @@ impl Trap {
         }
         Trap {
             inner: Arc::new(TrapInner {
-                reason: TrapReason::Message(message),
+                reason,
                 wasm_trace,
                 native_trace,
             }),
         }
     }
 
-    /// Returns a reference the `message` stored in `Trap`.
+    /// String representation the error stored in `Trap`.
     ///
     /// In the case of an explicit exit, the exit status can be obtained by
     /// calling `i32_exit_status`.
-    pub fn message(&self) -> &str {
+    pub fn message(&self) -> String {
```

Comment:
I think here it's best to leave this as `&str`. I don't think this is too too useful as an API and we can honestly probably remove it.

If this remains I think this should return a simple canned string for user-provided errors, since users of `Trap` need to be aware anyway that there's multiple ways to create a `Trap`.

Reply:
I've removed this fn since I agree it's not a useful API. I changed all tests using it to test for `trap.to_string().contains("...")`

Reply:
It seems that the most important place where this was used was the C api `wasm_trap_message`. I've switched that to `to_string()` if that's ok.